creator-node/src/routes/tracks.js

```diff
@@ -460,16 +461,22 @@ module.exports = function (app) {
   }))
 
   /**
-   * Gets a streamable mp3 link for a track by blockchainId. Supports range request headers.
+   * Gets a streamable mp3 link for a track by encodedId. Supports range request headers.
    * @dev - Wrapper around getCID, which retrieves track given its CID.
    **/
-  app.get('/tracks/stream/:blockchainId', async (req, res) => {
+  app.get('/tracks/stream/:encodedId', handleResponse(async (req, res) => {
```

Comment:
So I totally forgot to add this comment, but the files.js file which calls the same underlying method here doesn't call handleResponse explicitly so that the file data can be piped to the resp

* @dev This route does not handle responses by design, so we can pipe the gateway response.
   */
which was added because we do file.pipe(res) in streamFromFileSystem

But I see now that the errors thrown by this specific method are not handled particularly gracefully since it's not wrapped in a handle response.

Does this work for a normal request with handleResponse here? I suspect it doesn't, but maybe we can modify handleResponse slightly to accommodate the file piping use-case? wdyt?

Reply:
Discussed on Slack with both you and @SidSethi - this endpoint does work with handleResponse